src/utils/heading-handler.js

```diff
@@ -45,7 +45,7 @@ class HeadingHandler {
       // Inject the id after transformation
       const n = Object.assign({}, node);
       const el = fallback(h, n);
-      el.properties.id = headingIdentifier;
+      el.properties.id = el.properties.id || headingIdentifier;
```

Comment:
Just keep existing identifiers if defined by the author (not really possible in markdown for now, but if we use asciidoc as a parser at some point, then it would be supported